sailfishusermanager.cpp

```diff
@@ -309,6 +326,50 @@ int SailfishUserManager::removeUserFiles(uint uid)
     return rv;
 }
 
+/*
+ * Sets user quota limits, if supported by kernel and enabled on /home filesystem
+ */
+void SailfishUserManager::setUserLimits(uint uid)
+{
+    struct statvfs info;
+    errno = 0;
+    if (statvfs("/home", &info) < 0) {
+        qCWarning(lcSUM) << "Could not set limits, could not stat filesystem:" << strerror(errno);
+    } else {
+        // Soft limit of max(20 %, MAXIMUM_QUOTA_LIMIT), soft limit turns into hard after grace period
+        quint64 softlimit = ((quint64)info.f_bsize) * ((quint64)info.f_blocks) * 20 / 100;
+        if (softlimit > MAXIMUM_QUOTA_LIMIT)
+            softlimit = MAXIMUM_QUOTA_LIMIT;
+        // Hard limit is 120 % of soft limit
+        quint64 hardlimit = softlimit * 120 / 100;
+        qCDebug(lcSUM) << "Setting quota limits for" << uid << "to"
+                       << hardlimit << "and" << softlimit << "bytes";
+        // Sets block limits and clears inode limits
+        struct if_dqblk quota = {
+            .dqb_bhardlimit = (fsblkcnt_t)(hardlimit / BLOCK_SIZE),
+            .dqb_bsoftlimit = (fsblkcnt_t)(softlimit / BLOCK_SIZE),
+            .dqb_curspace = 0,
+            .dqb_ihardlimit = 0,
+            .dqb_isoftlimit = 0,
+            .dqb_curinodes = 0,
+            .dqb_btime = 0,
+            .dqb_itime = 0,
+            .dqb_valid = QIF_LIMITS
+        };
+        errno = 0;
+        if (quotactl(QCMD(Q_SETQUOTA, USRQUOTA), findHomeDevice().data(), (uid_t)uid, (caddr_t)&quota) < 0) {
```

Comment:
query hard coded "/home" vs set "/home/" obtained in roundabout way?

Reply:
Erm, no? Or I'm not sure what you mean but findHomeDevice() returns path to block device and not the mount point. That USER_HOME.arg("") elsewhere is maybe a little stupid though.

Reply:
To clarify: I meant that in this method we effetively arrive to "/home" via two ways: one is literal string, another one uses USER_HOME variable/constant -> works, but making also statfs() use value derived from USER_HOME would ensure they refer to the same thing also in the future.

Reply:
I see and yes unifying them would be better.